src/lib/panels/feedback.ts

```diff
@@ -0,0 +1,295 @@
+    // the panel or when the panel is closed programmatically)
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+
+    // Set the HTML content for the webview panel
+    this._panel.webview.html = this._getWebviewContent(
+      this._panel.webview,
+      extensionUri
+    );
+
+    // Set an event listener to listen for messages passed from the webview context
+    this._setWebviewMessageListener(this._panel.webview);
+  }
+
+  /**
+   * Renders the current webview panel if it exists otherwise a new webview panel
+   * will be created and displayed.
+   *
+   * @param extensionUri The URI of the directory containing the extension.
+   */
+  public static render(extensionUri: Uri) {
+    if (FeedbackPanel.currentPanel) {
+      // If the webview panel already exists reveal it
+      FeedbackPanel.currentPanel._panel.reveal(ViewColumn.One);
+    } else {
+      // If a webview panel does not already exist create and show a new one
+      const panel = window.createWebviewPanel(
+        // Panel view type
+        'feedbackPanel',
+        // Panel title
+        'HubSpot VSCode Extension Feedback',
+        // The editor column the panel should be displayed in
+        ViewColumn.One,
+        // Extra panel configurations
+        {
+          // Enable JavaScript in the webview
+          enableScripts: true,
+        }
+      );
+
+      FeedbackPanel.currentPanel = new FeedbackPanel(panel, extensionUri);
+    }
+  }
+
+  /**
+   * Cleans up and disposes of webview resources when the webview panel is closed.
+   */
+  public dispose() {
+    FeedbackPanel.currentPanel = undefined;
+
+    // Dispose of the current webview panel
+    this._panel.dispose();
+
+    // Dispose of all disposables (i.e. commands) for the current webview panel
+    while (this._disposables.length) {
+      const disposable = this._disposables.pop();
+      if (disposable) {
+        disposable.dispose();
+      }
+    }
+  }
+
+  /**
+   * Defines and returns the HTML that should be rendered within the webview panel.
+   *
+   * @remarks This is also the place where references to CSS and JavaScript files/packages
+   * (such as the Webview UI Toolkit) are created and inserted into the webview HTML.
+   *
+   * @param webview A reference to the extension webview
+   * @param extensionUri The URI of the directory containing the extension
+   * @returns A template string literal containing the HTML that should be
+   * rendered within the webview panel
+   */
+  private _getWebviewContent(webview: Webview, extensionUri: Uri) {
+    const mainUri = getUri(webview, extensionUri, [
+      'webview-ui',
+      'js',
+      'feedback-form.js',
+    ]);
+    const stylesheetUri = getUri(webview, extensionUri, [
+      'webview-ui',
+      'css',
+      'styles.css',
+    ]);
+    const userIdentificationInformation = getUserIdentificationInformation();
```

Comment:
Should we avoid collecting this if the user has telemetry disabled? Since there's indication we're tracking that here users might expect to be anonymized.

Reply:
Updated `getUserIdentificationInformation` to respect telemetry settings.